backends/arm/arm_tosa_backend.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2025 Arm Limited and/or its affiliates.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pyre-unsafe
+
+#
+# Main implementation of AoT flow to partition and preprocess for Arm target
+# backends. Converts via TOSA as an intermediate form supported by AoT and
+# JIT compiler flows.
+#
+import logging
+import os
+from typing import cast, final, List
+
+import serializer.tosa_serializer as ts  # type: ignore
+
+from executorch.backends.arm.arm_backend import get_tosa_spec
+from executorch.backends.arm.operators.node_visitor import get_node_visitors
+from executorch.backends.arm._passes.arm_pass_manager import (
+    ArmPassManager,
+)  # usort: skip
+from executorch.backends.arm.process_node import (
+    process_call_function,
+    process_output,
+    process_placeholder,
+)
+from executorch.backends.arm.tosa_utils import dbg_fail, dbg_tosa_dump
+from executorch.exir.backend.backend_details import BackendDetails, PreprocessResult
+from executorch.exir.backend.compile_spec_schema import CompileSpec
+from torch.export.exported_program import ExportedProgram
+from torch.fx import Node
+
+# TOSA backend debug functionality
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+TOSA_DBG_VERBOSE = os.environ.get("TOSA_DBG_VERBOSE") == "1"
```

Comment:
we need to do something like this across delegates, ideas welcome. I was going to follow pytorch/pytorch way of logging.